src/ESPEasyRules.ino

```diff
@@ -304,28 +304,84 @@ void replace_EventValueN_Argv(String& line, const String& argString, unsigned in
   }
+}
 
-  if (event.charAt(0) == '!') {
-    line.replace(F("%eventvalue%"), event); // substitute %eventvalue% with
-                                            // literal event string if
-                                            // starting with '!'
-  } else {
-    int equalsPos = event.indexOf("=");
+void process_strtol(String& line) {
+  // Syntax like 1234[strtol:16:38]7890
+  int startIndex;
+  while ((startIndex = line.indexOf(F("[strtol:"))) != -1) {    
+    String strolCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
+    int idxStart = strolCmd.indexOf(":") + 1;
+    int idxEnd = strolCmd.indexOf(":", idxStart);
+    String base = strolCmd.substring(idxStart, idxEnd);
+    idxStart = idxEnd + 1;
+    idxEnd = strolCmd.indexOf("]", idxStart);
+    String stringValue = strolCmd.substring(idxStart, idxEnd);
+       
+    line.replace(strolCmd, String(strtol(stringValue.c_str(), NULL, base.toInt())));   
+  }
+}
+
+void process_div100ths(String& line) {
+  // Syntax like XXX[div100ths:24:256]XXX
+  int startIndex;
+  while ((startIndex = line.indexOf(F("[div100ths:"))) != -1) {
+    String divCmd = line.substring(startIndex, line.indexOf("]", startIndex) + 1);
+    int idxStart = divCmd.indexOf(":") + 1;
+    int idxEnd = divCmd.indexOf(":", idxStart);
+    String dividend = divCmd.substring(idxStart, idxEnd);
+    idxStart = idxEnd + 1;
+    idxEnd = divCmd.indexOf("]", idxStart);
+    String divisor = divCmd.substring(idxStart, idxEnd);
+    float val = (100.0 * dividend.toInt()) / (1.0 * divisor.toInt());
+    char sval[10];
+    sprintf(sval, "%02d", (int)val);
+    line.replace(divCmd, String(sval));
+  }
+}
 
-    if (equalsPos > 0) {
-      // Replace %eventvalueX% with the actual value of the event.
-      // For compatibility reasons also replace %eventvalue%  (argc = 0)
-      String argString = event.substring(equalsPos + 1);
+void substitute_eventvalue(String& line, const String& event) {
+  if (line.indexOf(F("%eventvalue")) != -1) {
+    if (event.charAt(0) == '!') {
+      line.replace(F("%eventvalue%"), event); // substitute %eventvalue% with
+                                              // literal event string if
+                                              // starting with '!'
+    } else {
+      int equalsPos = event.indexOf("=");
 
-      for (unsigned int argc = 0; argc <= 4; ++argc) {
-        replace_EventValueN_Argv(line, argString, argc);
+      if (equalsPos > 0) {
+        // Replace %eventvalueX% with the actual value of the event.
+        // For compatibility reasons also replace %eventvalue%  (argc = 0)
+        String argString = event.substring(equalsPos + 1);
+
+        for (unsigned int argc = 0; argc <= 4; ++argc) {
+          replace_EventValueN_Argv(line, argString, argc);
+        }
       }
     }
   }
+
+  // process other markups as well
+  process_substring(line);
+  process_strtol(line);
+  process_div100ths(line);
```

Comment:
Just curious. Why do these markup parsing only in `substitute_eventvalue` and not in `parseTemplate` ?
https://github.com/letscontrolit/ESPEasy/blob/0c6e6ca915e5440de54213246dc8c1a8e6ca9cce/src/Misc.ino#L1538

The parseTemplate is also used on all other uses where we may address values from other tasks or system variables.

Reply:
To be honest? I downloaded the source of ESPEasy like the day I issued the pull request and still did not figure out the whole control flow.